### ISSUE TITLE
Extension package fails when using new (rebranded) category names

### ISSUE THREAD

**Issue by willsmythe** [Microsoft/tfs-cli#293]
See the following for the list of possible categories:
https://docs.microsoft.com/en-us/azure/devops/extend/develop/manifest?view=vsts#required-attributes

Package/publish commands fail when referencing any of these:
```
error: Error: There were errors with your extension. Address the following and re-run the tool.
error: The following categories are not valid: Pipelines, Repos, Artifacts, Boards. Valid categories are: Build and release, Collaborate, Code, Test, Plan and track, Insights, Integrate, Developer samples.
```

**Besides support these new categories**: since extensions targeting on-prem still need to use the older category names (since on-prem isn't aware of these new categories), we should detect and warn the developer if they are packaging an extension (not integration) that targets on-prem (e.g. target is MS.VS.S or MS.TF.S) and is using a new category.

**Comment by willsmythe** [comment]
@trevorsg - nice. I just updated yesterday, guess it was after the 0.6.3 push. Thanks.

**Comment by trevorsg** [comment]
@willsmythe I think you mean before :) But yeah, I published 0.6.3 yesterday.
